### PR TITLE
Prune runtime/net_policies.go

### PR DESCRIPTION
**Pull request by nebril**
Many of the tests covered by net_policies.go are already implemented in cilium-cli.

L7 HTTP method tests are ported in this CLI PR: https://github.com/cilium/cilium-cli/pull/1069 

Some of the more complex test cases were not removed, will discuss how to port these and get back to them soon.

### CODE REVIEW

File: test/runtime/net_policies.go [hunk 1]
```diff
@@ -35,14 +32,11 @@ const (
...
 
 	// Policy files
-	policyJSON                      = "policy.json"
-	multL7PoliciesJSON              = "Policies-l7-multiple.json"
-	policiesL7JSON                  = "Policies-l7-simple.json"
-	imposePoliciesL7JSON            = "Policies-l7-impose.json"
-	policiesL3JSON                  = "Policies-l3-policy.json"
-	policiesL4Json                  = "Policies-l4-policy.json"
-	policiesL3DependentL7EgressJSON = "Policies-l3-dependent-l7-egress.json"
-	policiesReservedInitJSON        = "Policies-reserved-init.json"
```
Comment (tklauser) [inline test/runtime/net_policies.go:45]: I guess the removed policy files could be removed from `test/manifests` as well, now that they're unused.

File: test/runtime/net_policies.go [hunk 2]
```diff
@@ -185,6 +185,7 @@ var _ = Describe("RuntimePolicies", func() {
 		}
 	}
 
+	// Cannot be tested in cilium-cli connectivity tests because of Daemon configuration changes
```
Comment (christarazi) [inline test/runtime/net_policies.go:91]: I wonder if these things are better tracked in a GH issue.

**Review by ldelossa — APPROVED** [review]

**Review by tklauser — APPROVED** [review]
LGTM apart from [Chris' comment](https://github.com/cilium/cilium/pull/21140#discussion_r960072319)

**Review by sayboras — APPROVED** [review]
LGTM :heavy_check_mark:

**Review by pchaigno — APPROVED** [review]

**Pull request merged**